pkg/controller/controller_binding.go

```diff
@@ -847,25 +845,21 @@ func (c *controller) injectServiceBinding(binding *v1beta1.ServiceBinding, crede
 		// Update existing secret
 		if !IsControlledBy(existingSecret, binding) {
 			controllerRef := GetControllerOf(existingSecret)
-			// TODO mark as terminal error once we have the terminal condition
```

Comment:
Why remove the TODO?

Reply:
Because the handling of terminal conditions / retries is already done now :)